assets/src/hooks/useChannel.ts

```diff
@@ -143,3 +143,82 @@ export const useCheckedChannel = <T, U>({
   ])
   return state
 }
+
+export const useCheckedTwoWayChannel = <T, U>({
+  socket,
+  topic,
+  event,
+  dataStruct,
+  parser,
+  loadingState,
+  initialMessage,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  initialMessage: any
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  loadingState: U
+}): [U, (event: string, payload: any) => void] => {
```

Comment:
**suggestion (non-blocking):** instead of `payload: any`, you could specify `payload: V` where `V` is a new parametrized type on the type signature of `useCheckedTwoWayChannel` - that way if another part of the code using this hook invokes the callback, it will enforce that it has to be used with a specific type.